lib/handlers/subscribe.js

```diff
@@ -65,6 +62,13 @@ function storeSubscriptions (sub, done) {
   var broker = client.broker
   var perst = broker.persistence
 
+  // total hack. Love it. Good enough for government work.  Probably should be redone with that TODO down there...
+  if (!sub) {
+    client.isAuthorizedForSub = false
+  } else {
+    client.isAuthorizedForSub = true
+  }
```

Comment:
This should not work with multiple parallel subscribes, we probably need the proper fix for this.

Reply:
I am seeing it now...yeah, this is going to require a lot more work :/